app/test/features/chat_ng/widgets/chat_messages_list_test.dart

```diff
@@ -0,0 +1,406 @@
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(emptyState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+          isActiveProvider(LabsFeature.chatUnread).overrideWith((ref) => false),
+          chat.hasUnreadMessages(testRoomId).overrideWith((ref) => false),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      await tester.pump();
+
+      // Now check for the list
+      expect(find.byType(AnimatedList), findsOneWidget);
+
+      expect(find.byType(ChatEvent), findsNothing);
+    });
+
+    testWidgets('renders messages correctly when available', (tester) async {
+      final messagesState = ChatRoomState(
+        messageList: List.generate(5, (index) => 'msg$index'),
+        loading: const ChatRoomLoadingState.loaded(),
+      );
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(messagesState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+          isActiveProvider(LabsFeature.chatUnread).overrideWith((ref) => false),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      await tester.pump();
+
+      expect(find.byType(AnimatedList), findsOneWidget);
+
+      //  verify the list has the correct number of items
+      final animatedList = tester.widget<AnimatedList>(
+        find.byType(AnimatedList),
+      );
+      expect(animatedList.initialItemCount, equals(5));
+    });
+
+    testWidgets('shows loading indicator when loading', (tester) async {
+      final loadingState = ChatRoomState(
+        messageList: ['msg1', 'msg2'],
+        loading: const ChatRoomLoadingState.loading(),
+      );
+
+      await tester.pumpProviderWidget(
+        overrides: [
+          chatMessagesStateProvider(
+            testRoomId,
+          ).overrideWith((ref) => MockChatRoomMessagesNotifier(loadingState)),
+          chat
+              .timelineStreamProvider(testRoomId)
+              .overrideWith((ref) => Future.value(MockTimelineStream())),
+          animatedListChatMessagesProvider(
+            testRoomId,
+          ).overrideWith((ref) => GlobalKey<AnimatedListState>()),
+        ],
+        child: ChatMessages(roomId: testRoomId),
+      );
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+    });
+
+    testWidgets('hides loading indicator when not loading', (tester) async {
```

Comment:
this is not what the tests checks for. It checks whether the loading animation isn't show _if it is started that way_. but I would like it to test what the test says instead: that it is shown in loading and when _changed_ switches to not showing. Maybe we can update the tests to check for that?

Reply:
changed the test scenario which now includes switches loading and not loading state while checking indicator state synchronously! 👍🏻